From 19665894b4787511ec08c43b51fd66c2ba912f94 Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Fri, 22 Nov 2024 17:11:27 +0100
Subject: [PATCH 1385/1695] plat-stm32mp2: add copro public key verify flag

The flag CFG_REMOTEPROC_PUB_KEY_VERIFY is disabled by default.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 89f40a60e4cfc77f4ca242ff195addd8be7f056a plat-stm32mp2: add copro public key verify flag]
Change-Id: I0cf81fb5818b5ecfbffd06efabe1329e7a7c33ce
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/420441
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 9da924b0c..3c186ab9d 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -259,6 +259,9 @@ CFG_REMOTEPROC_ENC_TESTKEY ?= n
 ifeq ($(CFG_REMOTEPROC_ENC_TESTKEY),y)
 $(call force,CFG_INSECURE,y,Required by CFG_REMOTEPROC_ENC_TESTKEY)
 endif
+
+# Co-processor public key verification against OTP fuses.
+CFG_REMOTEPROC_PUB_KEY_VERIFY ?= n
 endif
 
 # Default enable HWRNG PTA support
-- 
2.47.2

